app/src/main/java/com/example/hw_3/MainActivity.kt

```diff
@@ -0,0 +1,33 @@
+package com.example.hw_3
+
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import android.widget.Toolbar
+import androidx.navigation.findNavController
+import androidx.navigation.fragment.NavHostFragment
+import androidx.navigation.ui.AppBarConfiguration
+import androidx.navigation.ui.setupActionBarWithNavController
+import com.example.hw_3.R.id.toolbar
+import com.example.hw_3.databinding.ActivityMainBinding
+
+class MainActivity : AppCompatActivity() {
+
+    private lateinit var appBarConfiguration: AppBarConfiguration
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_main)
+
+        val host: NavHostFragment = supportFragmentManager
+            .findFragmentById(R.id.list) as NavHostFragment? ?: return
+
+        val navController = host.navController
+
+        appBarConfiguration = AppBarConfiguration(
+            setOf(R.id.fragment_details))
+
```

Comment:
посмотри семплы, тебе нужно было привязать тулбар к графу 
https://developer.android.com/guide/navigation/navigation-ui
```suggestion
        appBarConfiguration = AppBarConfiguration(navController.graph)
```